docs/tutorial.md

```diff
@@ -0,0 +1,241 @@
+- `bot.once(eventName,listener)`
+  Adds a one-time listener function for the event named `eventName`. The next time `eventName` is triggered, this listener is removed and then invoked. Basically it'll run one time for the first event.
+- `bot.removeListener(eventName,listener)`
+  Removes the specified listener from the listener array for the event named eventName.
+
+Not only bot object, [`Chest`](http://mineflayer.prismarine.js.org/#/api?id=mineflayerchest), [`Furnace`](http://mineflayer.prismarine.js.org/#/api?id=mineflayerfurnace), [`Dispenser`](http://mineflayer.prismarine.js.org/#/api?id=mineflayerdispenser), [`EnchantmentTable`](http://mineflayer.prismarine.js.org/#/api?id=mineflayerenchantmenttable), [`Villager`](http://mineflayer.prismarine.js.org/#/api?id=mineflayervillager) object also has its own events!
+
+Examples :
+
+#### Echo Bot
+
+In here we're listening [`chat`](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches) event, and resend the message.
+[`chat`](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches) event only emitted when a player chats publicly.
+
+```js
+const mineflayer = require('mineflayer');
+const options = {...}
+const bot = mineflayer.createBot(options);
+
+function echo(username, message) {
+  if (username === bot.username) return;
+  bot.chat(message);
+}
+
+bot.on('chat', echo); // Only emitted when a player chats publicly.
+```
+
+`bot.on('chat', echo);` will listend for [`chat`](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches) event, and execute `echo` function with argument of (username, message, translate, jsonMsg, matches) when [`chat`](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches) event triggered.
+
+#### Welcome Bot
+
+In here we're listening [`playerJoined`](http://mineflayer.prismarine.js.org/#/api?id=quotplayerjoinedquot-player) event.
+
+```js
+const mineflayer = require('mineflayer');
+const options = {...}
+const bot = mineflayer.createBot(options);
+
+function joined(player) {
+  bot.chat(`Welcome ${player}! :D`)
+}
+
+bot.on('playerJoined', joined)
+```
+
+### Callbacks
+
+Callback is a function as argument to the main function, when the main function finish or got an error then the callback executed.
+In [the API page](http://mineflayer.prismarine.js.org/#/api) callback abbreviated as `cb` or `callback`.
+Callback usually takes parameter called `err` and the second (if specified in) is a value that is passed to the callback, if `err` populated then error must be happened in the main function.
+
+Examples :
+
+#### Craft Stick Bot
+
+You don't wanna craft sticks, before the logs crafted into planks.
```

Comment:
Explain it more thoroughly. Random guy wouldn't understand why this sentence is even there.

Reply:
Got it